### DISCUSSION TITLE
Is it possible to implement UFCS with rhombus' macro system?

### DISCUSSION THREAD

**Discussion by glyh** [racket/rhombus#340] [Q&A] [answered]
This is a little bit off topic but the question is on the honu macro system. 

Bascially what I want to implement is an operator dot `.`, satisfying:

```
a.b(c, d, e) == b(a, c, d, e)
```

I saw https://github.com/racket/rhombus-prototype/discussions/296 requiring LHS to be an expression. I don't have an issue here, but rhs must be a function call. Will that be a problem?

**Comment by usaoc** [comment, marked as answer]
I don’t know if you have already figured this out yourself, but basically it depends on how complicated you want it to be.  The limitation is that the left-hand side of `.` will always be a parsed expression, but the right-hand side can be something unparsed.  A really dumb version:

```
#lang rhombus/and_meta
expr.macro '$left . $fn ($arg, ...)':
  '$fn #%call ($left, $arg, ...)'
```

This is dumb, because the expression following `.` has to be a term, but I think UFCS already has this limitation anyway.
